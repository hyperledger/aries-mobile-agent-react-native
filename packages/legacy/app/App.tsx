import {
  AgentProvider,
  // Agent,
  TourProvider,
  CommonUtilProvider,
  AuthProvider,
  ConfigurationProvider,
  NetworkProvider,
  StoreProvider,
  ThemeProvider,
  theme,
  initLanguages,
  initStoredLanguage,
  translationResources,
  ErrorModal,
  toastConfig,
  RootStack,
  NetInfo,
  defaultConfiguration,
  TourStep,
  TourBox,
  useTour,
  useTheme,
} from 'aries-bifold'
import * as React from 'react'
import { useEffect, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { StatusBar, Text } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

initLanguages(translationResources)

const App = () => {
  const { t } = useTranslation()

  useMemo(() => {
    initStoredLanguage().then()
  }, [])

  useEffect(() => {
    // Hide the native splash / loading screen so that our
    // RN version can be displayed.
    SplashScreen.hide()
  }, [])

  const tourSteps: TourStep[] = [
    {
      onBackdropPress: 'stop',
      render: () => {
        const { next, stop } = useTour()
        const { ColorPallet, TextTheme } = useTheme()
        return (
          <TourBox
            title={t('Tour.AddAndShare')}
            leftText={t('Tour.Skip')}
            rightText={t('Tour.Next')}
            onLeft={stop}
            onRight={next}
          >
            <Text
              style={{
                ...TextTheme.normal,
                color: ColorPallet.notification.infoText,
              }}
            >
              {t('Tour.AddAndShareDescription')}
            </Text>
          </TourBox>
        )
      },
    },
    {
      onBackdropPress: 'stop',
      render: () => {
        const { next, previous } = useTour()
        const { ColorPallet, TextTheme } = useTheme()
        return (
          <TourBox
            title={t('Tour.Notifications')}
            leftText={t('Tour.Back')}
            rightText={t('Tour.Next')}
            onLeft={previous}
            onRight={next}
          >
            <Text
              style={{
                ...TextTheme.normal,
                color: ColorPallet.notification.infoText,
              }}
            >
              {t('Tour.NotificationsDescription')}
            </Text>
          </TourBox>
        )
      },
    },
    {
      onBackdropPress: 'stop',
      render: () => {
        const { previous, stop } = useTour()
        const { ColorPallet, TextTheme } = useTheme()
        return (
          <TourBox
            title={t('Tour.YourCredentials')}
            leftText={t('Tour.Back')}
            rightText={t('Tour.Done')}
            onLeft={previous}
            onRight={stop}
          >
            <Text
              style={{
                ...TextTheme.normal,
                color: ColorPallet.notification.infoText,
              }}
            >
              {t('Tour.YourCredentialsDescription')}
            </Text>
          </TourBox>
        )
      },
    },
  ]

  return (
    <StoreProvider>
      <AgentProvider>
        <ThemeProvider value={theme}>
          <ConfigurationProvider value={defaultConfiguration}>
            <CommonUtilProvider>
              <AuthProvider>
                <NetworkProvider>
                  <StatusBar
                    hidden={false}
                    barStyle="light-content"
                    backgroundColor={theme.ColorPallet.brand.primary}
                    translucent={false}
                  />
                  <NetInfo />
                  <ErrorModal />
                  <TourProvider steps={tourSteps} overlayColor={'gray'} overlayOpacity={0.8}>
                    <RootStack />
                  </TourProvider>
                  <Toast topOffset={15} config={toastConfig} />
                </NetworkProvider>
              </AuthProvider>
            </CommonUtilProvider>
          </ConfigurationProvider>
        </ThemeProvider>
      </AgentProvider>
    </StoreProvider>
  )
}

export default App
