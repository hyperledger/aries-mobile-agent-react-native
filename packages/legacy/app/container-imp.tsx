import { Container, TokenMapping } from '@hyperledger/aries-bifold-core'
import { DependencyContainer } from 'tsyringe'

export class AppContainer implements Container {
  private _container: DependencyContainer

  public constructor(bifoldContainer: Container) {
    this._container = bifoldContainer.container.createChildContainer()
  }

  public get container(): DependencyContainer {
    return this._container
  }

  public init(): Container {
    // eslint-disable-next-line no-console
    console.log(`Initializing App container`)
    // Here you can register any component to override components in core package
    // Example: Replacing button in core with custom button
    // this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
    return this
  }

  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    return this.container.resolve(token) as TokenMapping[K]
  }

  public resolveSome = <K extends keyof TokenMapping>(tokens: K[]): TokenMapping[K][] => {
    return tokens.map((token) => this.resolve(token))
  }
}
