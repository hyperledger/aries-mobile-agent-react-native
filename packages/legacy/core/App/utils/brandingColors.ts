import { IColorPallet } from '../theme'
import { Shade, shadeIsLightOrDark } from './luminance'

export const backgroundColorIfErrorState = (
  ColorPallet: IColorPallet,
  error: boolean,
  predicateError: boolean,
  isProofRevoked: boolean,
  backgroundColor?: string
) => (error || predicateError || isProofRevoked ? ColorPallet.notification.errorBorder : backgroundColor)

export const fontColorWithHighContrast = (
  ColorPallet: IColorPallet,
  error: boolean,
  predicateError: boolean,
  isProofRevoked: boolean,
  backgroundColor?: string,
  proof?: boolean
) => {
  if (proof) {
    return ColorPallet.grayscale.mediumGrey
  }

  const c =
    backgroundColorIfErrorState(ColorPallet, error, predicateError, isProofRevoked, backgroundColor) ??
    ColorPallet.grayscale.lightGrey
  const shade = shadeIsLightOrDark(c)

  return shade == Shade.Light ? ColorPallet.grayscale.darkGrey : ColorPallet.grayscale.lightGrey
}
