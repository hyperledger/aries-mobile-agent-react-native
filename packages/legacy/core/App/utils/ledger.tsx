import TcpSocket from 'react-native-tcp-socket'

export const canConnectToLedgerNode = async (node: { host: string; port: number }): Promise<boolean> =>
  new Promise((resolve, reject) => {
    const socketTimeoutInMs = 3000
    const client = TcpSocket.createConnection(node, () => {
      resolve(true)
      client.destroy()
    })

    // Other events that can be safely be ignored. See the
    // library for more details:
    // https://www.npmjs.com/package/react-native-tcp-socket

    client.on('error', () => {
      client.destroy()
      reject(new Error('Error opening connection to ledger node'))
    })

    client.on('timeout', () => {
      client.destroy()
      client.removeAllListeners()

      reject(new Error('Timeout opening connection to ledger node'))
    })

    client.setTimeout(socketTimeoutInMs)
  })
