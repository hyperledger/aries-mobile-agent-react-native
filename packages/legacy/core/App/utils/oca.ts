import { CredentialExchangeRecord } from '@aries-framework/core'
import { legacy } from '@hyperledger/aries-oca'

import { CredentialSharedProofData, AnonCredsProofRequestTemplatePayloadData } from '../../verifier'

export const buildFieldsFromAnonCredsCredential = (credential: CredentialExchangeRecord): Array<legacy.Field> => {
  return credential?.credentialAttributes?.map((attr) => new legacy.Attribute(attr)) || []
}

export const buildFieldsFromAnonCredsProofRequestTemplate = (
  data: AnonCredsProofRequestTemplatePayloadData
): Array<legacy.Field> => {
  const fields = []
  if (data.requestedAttributes) {
    for (const item of data.requestedAttributes) {
      if (item.name) {
        fields.push(new legacy.Attribute({ name: item.name, value: null, ...item }))
      }
      if (item.names) {
        for (const name of item.names) {
          fields.push(new legacy.Attribute({ name, value: null, ...item }))
        }
      }
    }
  }
  if (data.requestedPredicates) {
    for (const item of data.requestedPredicates) {
      fields.push(
        new legacy.Predicate({
          pType: item.predicateType,
          pValue: item.predicateValue,
          ...item,
        })
      )
    }
  }
  return fields
}

export const buildFieldsFromSharedAnonCredsProof = (data: CredentialSharedProofData): Array<legacy.Field> => {
  const fields = []
  for (const attribute of data.sharedAttributes) {
    fields.push(
      new legacy.Attribute({
        name: attribute.name,
        value: attribute.value,
      })
    )
  }
  for (const attributesGroup of data.sharedAttributeGroups) {
    for (const attribute of attributesGroup.attributes) {
      fields.push(
        new legacy.Attribute({
          name: attribute.name,
          value: attribute.value,
        })
      )
    }
  }
  for (const predicate of data.resolvedPredicates) {
    fields.push(
      new legacy.Predicate({
        name: predicate.name,
        pType: predicate.predicateType,
        pValue: predicate.predicateValue,
      })
    )
  }
  return fields
}
