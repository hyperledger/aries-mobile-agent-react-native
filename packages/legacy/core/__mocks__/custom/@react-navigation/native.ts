const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()
const navigate = jest.fn()

const navigation = {
  navigate,
  setOptions: jest.fn(),
  getParent: jest.fn(() => ({
    navigate,
  })),
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { navigate, useNavigation, useIsFocused, useFocusEffect, createNavigatorFactory }
