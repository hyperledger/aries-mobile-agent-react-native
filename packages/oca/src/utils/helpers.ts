/**
 * Generates a numerical hash based on a given string
 * @see https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript
 * @param { string } s given string
 * @returns { number } numerical hash value
 */
export const hashCode = (s: string): number => {
  return s.split('').reduce((hash, char) => char.charCodeAt(0) + ((hash << 5) - hash), 0)
}

/**
 * Generates a pseudorandom number between 0 and 1 based on a seed
 * @see https://gist.github.com/tommyettinger/46a874533244883189143505d203312c
 * @see https://stackoverflow.com/questions/521295/seeding-the-random-number-generator-in-javascript
 * @param { number } seed any number
 * @returns { number } pseudorandom number between 0 and 1
 */
const mulberry32 = (seed: number) => {
  let t = (seed += 0x6d2b79f5)
  t = Math.imul(t ^ (t >>> 15), t | 1)
  t ^= t + Math.imul(t ^ (t >>> 7), t | 61)
  return ((t ^ (t >>> 14)) >>> 0) / 4294967296
}

/**
 * Converts a numerical hash into a hexidecimal color string
 * @see https://helderesteves.com/generating-random-colors-js/#Generating_random_dark_colors
 * @param { number } hash numerical hash value (generated by hashCode function above)
 * @returns { string } hexidecimal string eg. #32d3cc
 */
export const hashToRGBA = (hash: number) => {
  let color = '#'
  const colorRangeUpperBound = 256

  // once for r, g, b
  for (let i = 0; i < 3; i++) {
    // append a pseudorandom two-char hexidecimal value from the lower half of the color spectrum (to limit to darker colors)
    color += ('0' + Math.floor((mulberry32(hash + i) * colorRangeUpperBound) / 2).toString(16)).slice(-2)
  }

  return color
}
