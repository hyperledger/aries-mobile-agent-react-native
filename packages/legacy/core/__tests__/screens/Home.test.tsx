import {
  BasicMessageRecord,
  BasicMessageRole,
  CredentialExchangeRecord as CredentialRecord,
  CredentialRole,
  CredentialState,
  ProofExchangeRecord,
  ProofRole,
  ProofState,
} from '@credo-ts/core'
import { useNavigation } from '@react-navigation/native'
import { fireEvent, render, waitFor, act } from '@testing-library/react-native'
// import { useAgent } from '@credo-ts/react-hooks'

import React, { useMemo } from 'react'

// eslint-disable-next-line import/no-named-as-default
import Home from '../../App/screens/Home'
import { testIdWithKey } from '../../App/utils/testable'
import { BasicAppContext } from '../helpers/app'
import timeTravel from '../helpers/timetravel'

import { useBasicMessages, useCredentialByState, useProofByState } from '@credo-ts/react-hooks'

jest.useFakeTimers()

describe('displays a home screen', () => {
  beforeEach(() => {
    jest.clearAllTimers()
    jest.clearAllMocks()
    // jest.resetAllMocks()
  })

  test.skip('defaults to no notifications', async () => {
    // @ts-ignore-next-line
    useBasicMessages.mockReturnValue({ records: [] as BasicMessageRecord[] })
    // @ts-ignore-next-line
    useCredentialByState.mockImplementation((state: string) => {
      return useMemo(() => [] as CredentialRecord[], [state])
    })
    // @ts-ignore-next-line
    // useProofByState.mockImplementation((state: string) => {
    //   return useMemo(() => [] as ProofExchangeRecord[], [state])
    // })
    // @ts-ignore-next-line
    useProofByState.mockReturnValue([] as ProofExchangeRecord[])

    const tree = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    await act(async () => {
      jest.runAllTimers()

      // const notificationLabel = await tree.findByText(testIdWithKey('NoNewUpdates'))
      const notificationLabel = await tree.findByTestId(testIdWithKey('NoNewUpdates'))
      expect(tree).toMatchSnapshot()
      expect(notificationLabel).toBeTruthy()
    })
  })

  test.skip('notifications are displayed', async () => {
    const tree = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    await act(async () => {
      jest.runAllTimers()
      const flatListInstance = await tree.findAllByTestId(testIdWithKey('NotificationListItem'))
      expect(flatListInstance).toHaveLength(5)
    })
  })

  test('touch offer navigation correctly', async () => {
    const navigation = useNavigation()
    const view = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    const button = await view.findByTestId(testIdWithKey('ViewOffer'))

    expect(button).toBeDefined()

    await act(() => {
      fireEvent(button, 'press')
    })

    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Connection Stack', {
      screen: 'Connection',
      params: { credentialId: '99bbf805-fc82-44dc-82eb-e3eb1e7f8ab7' },
    })
  })

  test('touch proof navigates correctly', async () => {
    const { findAllByTestId } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    const buttons = await findAllByTestId(testIdWithKey('ViewProofRecord'))
    const navigation = useNavigation()

    expect(buttons).toBeDefined()

    await act(() => {
      fireEvent(buttons[0], 'press')

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('Connection Stack', {
        screen: 'Connection',
        params: { proofId: '5658ee3f-04dc-487a-8524-3e070c820b8e' },
      })
    })
  })

  test('touch basic message navigation correctly', async () => {
    const navigation = useNavigation()
    const { findAllByTestId } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    await act(async () => {
      const button = (await findAllByTestId(testIdWithKey('ViewBasicMessage')))[0]
      expect(button).toBeDefined()
      fireEvent(button, 'press')

      expect(navigation.getParent()?.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.getParent()?.navigate).toHaveBeenCalledWith('Contacts Stack', {
        screen: 'Chat',
        params: { connectionId: 'c426f2dc-0ffc-4252-b7d6-2304755f84d9' },
      })
    })
  })
})
