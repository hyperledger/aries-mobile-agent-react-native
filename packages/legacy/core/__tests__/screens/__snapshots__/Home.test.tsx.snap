// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "_tags": Object {},
        "autoAcceptProof": undefined,
        "connectionId": undefined,
        "createdAt": "2024-09-05T18:56:08.770Z",
        "errorMessage": undefined,
        "id": "5658ee3f-04dc-487a-8524-3e070c820b8e",
        "isVerified": undefined,
        "metadata": Object {},
        "parentThreadId": "d06b0c33-ba17-42d7-9334-afdf60403f02",
        "protocolVersion": "v1",
        "role": "prover",
        "state": "request-received",
        "threadId": "dbc57c37-63b8-40ed-bb14-4b58e86db4ec",
      },
      Object {
        "_tags": Object {
          "connectionId": "20f1f732-b64f-4d52-99fd-e13fe0d9e62f",
          "role": "prover",
          "state": "request-received",
          "threadId": "985a0d98-bc31-435c-a3fe-2f884acae4fe",
        },
        "autoAcceptProof": undefined,
        "connectionId": "20f1f732-b64f-4d52-99fd-e13fe0d9e62f",
        "createdAt": "2024-09-04T21:12:14.072Z",
        "errorMessage": undefined,
        "id": "c54dfe4e-925d-4b9a-9f2c-2adeb308c5de",
        "isVerified": undefined,
        "metadata": Object {},
        "parentThreadId": undefined,
        "protocolVersion": "v1",
        "role": "prover",
        "state": "request-received",
        "threadId": "985a0d98-bc31-435c-a3fe-2f884acae4fe",
      },
      Object {
        "_tags": Object {
          "connectionId": "efa7d36e-9dbe-4c0b-b128-556e731d329a",
          "role": "prover",
          "state": "request-received",
          "threadId": "78de35dd-980e-4379-a668-3a6e9c4365d4",
        },
        "autoAcceptProof": undefined,
        "connectionId": "efa7d36e-9dbe-4c0b-b128-556e731d329a",
        "createdAt": "2024-09-04T21:12:14.072Z",
        "errorMessage": undefined,
        "id": "bbbf9c83-7930-4c97-944f-9b6adbbc8f49",
        "isVerified": undefined,
        "metadata": Object {},
        "parentThreadId": undefined,
        "protocolVersion": "v1",
        "role": "prover",
        "state": "request-received",
        "threadId": "78de35dd-980e-4379-a668-3a6e9c4365d4",
      },
      Object {
        "_tags": Object {},
        "connectionId": "c426f2dc-0ffc-4252-b7d6-2304755f84d9",
        "content": "Earl Grey",
        "createdAt": "2024-09-04T18:53:11.771Z",
        "id": "cbd8f4f0-36ce-4a2e-8c47-d9d5cbaa5617",
        "metadata": Object {},
        "parentThreadId": undefined,
        "role": "receiver",
        "sentTime": "20200303",
        "threadId": "18b5dad0-49d1-4e26-b52d-bb01ee5f4f53",
      },
      Object {
        "_tags": Object {},
        "autoAcceptCredential": undefined,
        "connectionId": "b4b1b9cd-a445-4bb3-9645-a2377471965a",
        "createdAt": "2024-09-04T18:50:29.771Z",
        "credentialAttributes": Array [
          Object {
            "mime-type": "text/plain",
            "name": "student_first_name",
            "value": "Alice",
          },
          Object {
            "mime-type": "text/plain",
            "name": "student_last_name",
            "value": "Smith",
          },
          Object {
            "mime-type": "text/plain",
            "name": "expiry_date",
            "value": "20280820",
          },
        ],
        "credentials": Array [],
        "errorMessage": undefined,
        "id": "99bbf805-fc82-44dc-82eb-e3eb1e7f8ab7",
        "linkedAttachments": undefined,
        "metadata": Object {
          "_anoncreds/credential": Object {
            "credentialDefinitionId": "QEquAHkM35w4XVT3Ku5yat:3:CL:834668:student_card",
            "schemaId": "QEquAHkM35w4XVT3Ku5yat:2:student_card:1.6",
          },
        },
        "parentThreadId": undefined,
        "protocolVersion": "v1",
        "revocationNotification": undefined,
        "role": "holder",
        "state": "offer-received",
        "threadId": "735df561-5346-4cb0-b11f-44a49984c0c3",
      },
    ]
  }
  decelerationRate="fast"
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginBottom": 35,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    >
      <View />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "marginLeft": 0,
                "marginRight": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î¢Ž
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 18,
                "fontWeight": "normal",
              }
            }
            testID="com.ariesbifold:id/NoNewUpdates"
          >
            Home.NoNewUpdates
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <View
          style={
            Object {
              "paddingBottom": 75,
              "paddingHorizontal": 25,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 25,
                  "marginTop": 35,
                },
              ]
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 38,
                    "fontWeight": "bold",
                  },
                  Object {
                    "marginBottom": 20,
                    "marginTop": 25,
                  },
                ]
              }
            >
              Home.Welcome
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 25,
                  "marginTop": 35,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 25,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Home.NoCredentials
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`with a notifications module, when an issuer sends a credential offer notifications are displayed 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "_tags": Object {},
        "autoAcceptProof": undefined,
        "connectionId": undefined,
        "createdAt": "2024-09-05T18:56:08.770Z",
        "errorMessage": undefined,
        "id": "5658ee3f-04dc-487a-8524-3e070c820b8e",
        "isVerified": undefined,
        "metadata": Object {},
        "parentThreadId": "d06b0c33-ba17-42d7-9334-afdf60403f02",
        "protocolVersion": "v1",
        "role": "prover",
        "state": "request-received",
        "threadId": "dbc57c37-63b8-40ed-bb14-4b58e86db4ec",
      },
      Object {
        "_tags": Object {
          "connectionId": "20f1f732-b64f-4d52-99fd-e13fe0d9e62f",
          "role": "prover",
          "state": "request-received",
          "threadId": "985a0d98-bc31-435c-a3fe-2f884acae4fe",
        },
        "autoAcceptProof": undefined,
        "connectionId": "20f1f732-b64f-4d52-99fd-e13fe0d9e62f",
        "createdAt": "2024-09-04T21:12:14.072Z",
        "errorMessage": undefined,
        "id": "c54dfe4e-925d-4b9a-9f2c-2adeb308c5de",
        "isVerified": undefined,
        "metadata": Object {},
        "parentThreadId": undefined,
        "protocolVersion": "v1",
        "role": "prover",
        "state": "request-received",
        "threadId": "985a0d98-bc31-435c-a3fe-2f884acae4fe",
      },
      Object {
        "_tags": Object {
          "connectionId": "efa7d36e-9dbe-4c0b-b128-556e731d329a",
          "role": "prover",
          "state": "request-received",
          "threadId": "78de35dd-980e-4379-a668-3a6e9c4365d4",
        },
        "autoAcceptProof": undefined,
        "connectionId": "efa7d36e-9dbe-4c0b-b128-556e731d329a",
        "createdAt": "2024-09-04T21:12:14.072Z",
        "errorMessage": undefined,
        "id": "bbbf9c83-7930-4c97-944f-9b6adbbc8f49",
        "isVerified": undefined,
        "metadata": Object {},
        "parentThreadId": undefined,
        "protocolVersion": "v1",
        "role": "prover",
        "state": "request-received",
        "threadId": "78de35dd-980e-4379-a668-3a6e9c4365d4",
      },
      Object {
        "_tags": Object {},
        "connectionId": "c426f2dc-0ffc-4252-b7d6-2304755f84d9",
        "content": "Earl Grey",
        "createdAt": "2024-09-04T18:53:11.771Z",
        "id": "cbd8f4f0-36ce-4a2e-8c47-d9d5cbaa5617",
        "metadata": Object {},
        "parentThreadId": undefined,
        "role": "receiver",
        "sentTime": "20200303",
        "threadId": "18b5dad0-49d1-4e26-b52d-bb01ee5f4f53",
      },
      Object {
        "_tags": Object {},
        "autoAcceptCredential": undefined,
        "connectionId": "b4b1b9cd-a445-4bb3-9645-a2377471965a",
        "createdAt": "2024-09-04T18:50:29.771Z",
        "credentialAttributes": Array [
          Object {
            "mime-type": "text/plain",
            "name": "student_first_name",
            "value": "Alice",
          },
          Object {
            "mime-type": "text/plain",
            "name": "student_last_name",
            "value": "Smith",
          },
          Object {
            "mime-type": "text/plain",
            "name": "expiry_date",
            "value": "20280820",
          },
        ],
        "credentials": Array [],
        "errorMessage": undefined,
        "id": "99bbf805-fc82-44dc-82eb-e3eb1e7f8ab7",
        "linkedAttachments": undefined,
        "metadata": Object {
          "_anoncreds/credential": Object {
            "credentialDefinitionId": "QEquAHkM35w4XVT3Ku5yat:3:CL:834668:student_card",
            "schemaId": "QEquAHkM35w4XVT3Ku5yat:2:student_card:1.6",
          },
        },
        "parentThreadId": undefined,
        "protocolVersion": "v1",
        "revocationNotification": undefined,
        "role": "holder",
        "state": "offer-received",
        "threadId": "735df561-5346-4cb0-b11f-44a49984c0c3",
      },
    ]
  }
  decelerationRate="fast"
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginBottom": 35,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    >
      <View />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "marginLeft": 0,
                "marginRight": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î¢Ž
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 18,
                "fontWeight": "normal",
              }
            }
            testID="com.ariesbifold:id/NoNewUpdates"
          >
            Home.NoNewUpdates
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <View
          style={
            Object {
              "paddingBottom": 75,
              "paddingHorizontal": 25,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 25,
                "marginTop": 35,
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 38,
                    "fontWeight": "bold",
                  },
                  Object {
                    "marginBottom": 20,
                    "marginTop": 25,
                  },
                ]
              }
            >
              Home.Welcome
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 25,
                "marginTop": 35,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 25,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Home.NoCredentials
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
